x/evm/keeper/chainKeeper.go

```diff
@@ -91,12 +90,13 @@ func (k chainKeeper) getCommandsGasLimit(ctx sdk.Context) uint32 {
 	return commandsGasLimit
 }
 
-func (k chainKeeper) GetChainID(ctx sdk.Context) (*big.Int, bool) {
+func (k chainKeeper) GetChainID(ctx sdk.Context) (sdk.Int, bool) {
 	network, ok := k.GetNetwork(ctx)
 	if !ok {
-		return nil, false
+		return sdk.Int{}, false
 	}
-	return k.GetChainIDByNetwork(ctx, network), true
+
+	return k.GetChainIDByNetwork(ctx, network)
```

Comment:
Btw this was the other missing validation check before, `GetChainIDByNetwork` could have returned `nil`, but we always returned `true`